### PR TITLE
refactor: fix scene controller singleton references (III)

### PR DESCRIPTION
**Pull request by BrianAmadori**
https://play.decentraland.zone/branch/refactor/fix-scene-controller-singleton-references/index.html?ENV=org

### CODE REVIEW

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldController.cs [hunk 1]
```diff
@@ -625,21 +647,25 @@ public VoxelEntityHit GetCloserUnselectedVoxelEntityOnPointer()
...
         {
             string entityID = hit.collider.gameObject.name;
 
             if (sceneToEdit.entities.ContainsKey(entityID))
             {
                 DCLBuilderInWorldEntity entityToCheck = builderInWorldEntityHandler.GetConvertedEntity(sceneToEdit.entities[entityID]);
+
                 if (entityToCheck == null) continue;
 
+                Camera camera = Camera.main;
```
Comment (sandrade-dcl) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldController.cs:661]: I think it's better to declare de `Camera` variable as a private variable of the class and move this assignation out of the loop (`Awake`, `Start`...?) in order to execute it only once time, don't you?

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/MessagingBus.cs [hunk 2]
```diff
@@ -116,7 +116,7 @@ public MessagingBus(MessagingBusType type, IMessageProcessHandler handler, Messa
             this.type = type;
             this.owner = owner;
             this.pendingMessagesCount = 0;
-            sceneController = SceneController.i;
+            sceneController = Environment.i.sceneController;
```
Comment (sandrade-dcl) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/MessagingBus.cs]: I notice that this `sceneController` private variable is not being used in this class (only to be assigned in the constructor). We should remove it.

**Review by sandrade-dcl — CHANGES_REQUESTED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/MessagingBus.cs (hunk 2 above)
Comment (BrianAmadori) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/MessagingBus.cs]: Will do!

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldController.cs (hunk 1 above)
Comment (BrianAmadori) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldController.cs:661]: This actually falls out of the scope of this PR. I just couldn't bear the repeated use of `Camera.main` and and kinda mitigated it. We should make this a private member for sure, but I think its better to do it in another PR.

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/MessagingBus.cs (hunk 2 above)
Comment (sandrade-dcl) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/MessagingBus.cs]: Thanks! The only thing left to do is to make this change, so I'm going to approve the PR.

**Review by sandrade-dcl — APPROVED** [review]

**Review by AjimenezDCL — APPROVED** [review]
Good job!

File: unity-client/Assets/Scripts/MainScripts/Debugging/WSS/WSSController.cs [hunk 3]
```diff
@@ -279,26 +278,32 @@ void Update()
...
+                                SceneController.i.UnloadAllScenesQueued();
+                                break;
+                            case "CreateUIScene":
+                                SceneController.i.CreateUIScene(msg.payload);
+                                break;
+                            case "BuilderReady":
+                                Main.i.BuilderReady();
+                                break;
+                            case "UpdateParcelScenes":
+                                SceneController.i.UpdateParcelScenes(msg.payload);
```
Comment (pravusjif) [inline unity-client/Assets/Scripts/MainScripts/Debugging/WSS/WSSController.cs]: all of the `SceneController.i` used here should use `Environment.i.sceneController` right?

**Review by pravusjif — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/MessagingBus.cs (hunk 2 above)
Comment (BrianAmadori) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/MessagingBus.cs]: feedback applied!

File: unity-client/Assets/Scripts/MainScripts/Debugging/WSS/WSSController.cs (hunk 3 above)
Comment (BrianAmadori) [reply unity-client/Assets/Scripts/MainScripts/Debugging/WSS/WSSController.cs]: Yep, something went wrong in a merge, ty for noticing. Fixed.

**Pull request merged**